ethers-providers/src/rpc/transports/common.rs

```diff
@@ -109,120 +109,122 @@ pub struct Params<'a> {
     pub result: &'a RawValue,
 }
 
-// FIXME: ideally, this could be auto-derived as an untagged enum, but due to
-// https://github.com/serde-rs/serde/issues/1183 this currently fails
-impl<'de: 'a, 'a> Deserialize<'de> for Response<'a> {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+struct ResponseVisitor<'b,const RELAXED:bool>(&'b ());
```

Comment:
This should use `PhantomData`

```suggestion
struct ResponseVisitor<'b,const RELAXED:bool>(PhantomData<&'b ()>);
```